File: Leetcode/BuildArray/BuildArray.cpp
#include <vector>
#include <string>

using namespace std;

class Solution
{
public:
    vector<string> buildArray(vector<int> &target, int n)
    {
        vector<string> res;
        for (int i = 0, j = 1; i < target.size() && j <= n; j++)
        {
            if (j == target[i])
            {
                res.push_back("Push");
                i++;
            }
            else
            {
                res.push_back("Push");
                res.push_back("Pop");
            }
        }
        return res;
    }
};

File: Leetcode/Leetcode9/BuildArray.java
package Leetcode.Leetcode9;

import java.util.ArrayList;
import java.util.List;

public class BuildArray {
    public List<String> buildArray(int[] target, int n) {
        List<String> ans = new ArrayList<>();
        for (int i = 1, temp = 0; i <= n && temp < target.length; i++) {
            if (target[temp] == i) {
                ans.add("Push");
                temp++;
            } else {
                ans.add("Push");
                ans.add("Pop");
            }
        }
        return ans;
    }
}
